src/Arcus.Observability.Telemetry.Serilog.Filters/TelemetryTypeFilter.cs

```diff
@@ -68,16 +68,30 @@ public bool IsEnabled(LogEvent logEvent)
             switch (TelemetryType)
             {
                 case TelemetryType.Dependency:
-                    return logEvent.Properties.ContainsKey(ContextProperties.DependencyTracking.DependencyLogEntry);
+                    return IsFilteringRequired(ContextProperties.DependencyTracking.DependencyLogEntry, logEvent);
                 case TelemetryType.Request:
-                    return logEvent.Properties.ContainsKey(ContextProperties.RequestTracking.RequestLogEntry);
+                    return IsFilteringRequired(ContextProperties.RequestTracking.RequestLogEntry, logEvent);
                 case TelemetryType.Events:
-                    return logEvent.Properties.ContainsKey(ContextProperties.EventTracking.EventLogEntry);
+                    return IsFilteringRequired(ContextProperties.EventTracking.EventLogEntry, logEvent);
                 case TelemetryType.Metrics:
-                    return logEvent.Properties.ContainsKey(ContextProperties.MetricTracking.MetricLogEntry);
+                    return IsFilteringRequired(ContextProperties.MetricTracking.MetricLogEntry, logEvent);
+                case TelemetryType.Trace:
+                    return true; // TODO: How should we detect this?
```

Comment:
@stijnmoreels From what I'm seeing this is hard to identify, is there a better approach?

Reply:
True, even the official Azure Application Insights Serilog sink only delegates all log events as traces or events without any identifiable properties.
Hmm, normally, everything that is not above is considered trace, I think, so that's why it was set to `true`.

Reply:
I added that but let's keep it like that then!

One thing I'm wondering is, should I add a guard in On(...) since we don't support filtering on it?

Reply:
Ah, yeah, that's a good idea. Making it even more user-friendly that way.

Reply:
Done, please take another look!